File: ts/wam/7076.ts
export const WamBusinessTemplateRichOrderStatus = 7076 as const

export interface BusinessTemplateRichOrderStatusEvent {
  /** field 1, wire `action_type_rich_order_status` */
  actionTypeRichOrderStatus?: string
  /** field 2, wire `business_jid` */
  businessJid?: string
  /** field 3, wire `chats_folder_type` */
  chatsFolderType?: CHATSFOLDERTYPE
  /** field 4, wire `contact_type` */
  contactType?: CONTACTTYPE
  /** field 5, wire `is_biz_intent` */
  isBizIntent?: boolean
  /** field 6, wire `is_insub_contact` */
  isInsubContact?: boolean
  /** field 7, wire `is_muted` */
  isMuted?: boolean
  /** field 8, wire `read_receipts_enabled` */
  readReceiptsEnabled?: boolean
  /** field 9, wire `template_id` */
  templateId?: string
}

File: go/wam/7076.go
const WamBusinessTemplateRichOrderStatus = 7076 // channel: private

type BusinessTemplateRichOrderStatusEvent struct {
	ActionTypeRichOrderStatus string `wam:"action_type_rich_order_status"` // field 1
	BusinessJid string `wam:"business_jid"` // field 2
	ChatsFolderType CHATSFOLDERTYPE `wam:"chats_folder_type"` // field 3
	ContactType CONTACTTYPE `wam:"contact_type"` // field 4
	IsBizIntent bool `wam:"is_biz_intent"` // field 5
	IsInsubContact bool `wam:"is_insub_contact"` // field 6
	IsMuted bool `wam:"is_muted"` // field 7
	ReadReceiptsEnabled bool `wam:"read_receipts_enabled"` // field 8
	TemplateId string `wam:"template_id"` // field 9
}